src/Data/WideWord/Int128.hs

```diff
@@ -65,6 +64,14 @@ import Data.Primitive.Types (Prim (..), defaultSetByteArray#, defaultSetOffAddr#
 
 import Data.Hashable (Hashable,hashWithSalt)
 
+import Data.WideWord.Compat
```

Comment:
@erikd I've consolidated the `compat` stuff to minimize noise here. Thoughts?

Reply:
I like it. That is a significant improvement.

Reply:
The only thing I would still like to address is the `CPP` and `ViewPatterns` use in things like say `Data.WideWord.Word256.signum256`. I am sure there is a neater way to do that (in say that `Compat` module).

Reply:
Maybe something like:

```
signum256 :: Word256 -> Word256
signum256 (Word256 a b c d) =
  if isZeroWord64 a && isZeroWord64 b && isZeroWord64 c && isZeroWord64 d
    then zeroWord256
    else oneWord256
```
With the `isZeroWord64` being defined in the `Compat` module.

Reply:
Also wondering if some of the other CPPisms like this:
```
#if MIN_VERSION_base(4,17,0)
  Word256 (W64# (or64# a3 b3)) (W64# (or64# a2 b2))
          (W64# (or64# a1 b1)) (W64# (or64# a0 b0))
#else
  Word256 (W64# (or# a3 b3)) (W64# (or# a2 b2))
          (W64# (or# a1 b1)) (W64# (or# a0 b0))
#endif
```
could be lifted into the `Compat` module.

Reply:
This is all made more difficult because `ghc-9.4` insists on being built with the Hadrian build system which complete breaks the way I used to manage installing multiple GHC versions.

Reply:
Oh, totally - I only did the `Int128` to start since I didn't want to do a bunch of work if it wasn't to your tastes 😅  I'll port the rest over.

Reply:
also tbh `ghcup` does work very well, you may consider adopting it? i was pretty skeptical but it is legitimately very good

Reply:
Can `ghcup` produce a package I can install on several machines? Can I specify where it installs stuff like `/usr/lib/ghc-X.Y` ?